src/NexusMods.App/GlobalFlags.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Concurrent;
+
+namespace NexusMods.App;
+
+/// <summary>
+/// Holds global information about the application.
+/// </summary>
+public static class GlobalFlags
+{
+    /// <summary>
+    /// Holds the thread that started the application.
+    /// </summary>
+    public static Thread StartingThread { get; internal set; } = null!;
+
```

Comment:
This property should have some safeguards to prevent misuse. [`InvalidOperationException`](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception) is thrown when a method call is invalid for the object's current state.

```csharp
private static Thread? _startingThread = null;

public static Thread StartingThread
{
  get
  {
    if (_startingThread is null) throw new InvalidOperationException();
    return _startingThread;
  }
  set
  {
    if (_startingThread is not null) throw new InvalidOperationException();
    _startingThread = value;
  }
}
```

Reply:
Decided to simplify this even more and put it behind an argumentless method that sets the private thread method to the current thread.